### PR TITLE
Bug 769745 - pickle account

### PR DESCRIPTION
**Pull request by ncalexan**
Still needs Bug 735842 (https://bugzilla.mozilla.org/show_bug.cgi?id=735842) to hit m-i for this to be testable as is.  Review can definitely begin.

I still want to give this a thorough device test before landing.

### CODE REVIEW

File: src/main/java/org/mozilla/gecko/sync/Utils.java [hunk 1]
```diff
@@ -441,4 +445,46 @@ public static void putStageNamesToSync(final Bundle bundle, final String[] stage
...
   }
+
+  /**
+   * Read contents of file as a string.
+   *
+   * @param context Android context.
+   * @param filename name of file to read.
+   * @return <code>String</code> instance.
+   */
+  public static String readFile(final Context context, final String filename) {
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/sync/Utils.java]: Either document that `filename` is not null, or throw/check.

File: src/main/java/org/mozilla/gecko/sync/config/AccountPickler.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
+package org.mozilla.gecko.sync.config;
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/sync/config/AccountPickler.java:5]: MPL.

File: src/main/java/org/mozilla/gecko/sync/config/AccountPickler.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  public static void pickle(final Context context, final String filename,
+      final SyncAccountParameters params, final boolean syncAutomatically) {
+    final ExtendedJSONObject o = params.asJSON();
+    o.put(Constants.JSON_KEY_SYNC_AUTOMATICALLY, new Boolean(syncAutomatically));
+    o.put(Constants.JSON_KEY_VERSION, new Long(VERSION));
+    o.put(Constants.JSON_KEY_TIMESTAMP, new Long(System.currentTimeMillis()));
+
+    try {
+      final FileOutputStream fos = context.openFileOutput(filename, Context.MODE_PRIVATE);
+      final PrintStream ps = (new PrintStream(fos));
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/sync/config/AccountPickler.java]: Excess parens.

File: src/main/java/org/mozilla/gecko/sync/config/AccountPickler.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      ps.print(o.toJSONString());
+      ps.close();
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/sync/config/AccountPickler.java]: `finally`. Also, closing the `PrintStream` will close the underlying `FileOutputStream` for you.

File: src/main/java/org/mozilla/gecko/sync/setup/SyncAccounts.java [hunk 5]
```diff
@@ -139,11 +184,21 @@ public SyncAccountParameters(Context context, AccountManager accountManager,
    * occurred and the account could not be added.
    */
   public static class CreateSyncAccountTask extends AsyncTask<SyncAccountParameters, Void, Account> {
+    protected final boolean syncAutomatically;
+
+    public CreateSyncAccountTask() {
+      this.syncAutomatically = true;
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/sync/setup/SyncAccounts.java]:
```
  this(true);
```

File: test/src/org/mozilla/gecko/sync/config/test/TestAccountPickler.java [hunk 6]
```diff
@@ -0,0 +1,208 @@
+package org.mozilla.gecko.sync.config.test;
```
Comment (rnewman) [inline test/src/org/mozilla/gecko/sync/config/test/TestAccountPickler.java:4]: License.

**Comment by rnewman** [comment]
If this works on a device, consider me delighted. Land it on inbound with nits.

**Comment by ncalexan** [comment]
Re-pushed with nits addressed and a commit to preserve existing settings (device testing revealed problems here).  See

https://bugzilla.mozilla.org/show_bug.cgi?id=769745#c1

for... exhaustive... testing instructions.  Needs Bug 735842 to land for this to work.

**Comment by ncalexan** [comment]
Sad face: integration tests wipe any existing Sync accounts.  This can be worked around, but not right now.

**Pull request merged**